lib/src/repo.rs

```diff
@@ -462,6 +487,23 @@ impl StoreFactories {
         Ok(backend_factory(settings, store_path)?)
     }
 
+    pub fn load_signer(&self, settings: &UserSettings) -> Result<Signer, StoreLoadError> {
+        let config = settings.signing_backend();
+
+        let mut main_backend = None;
+        let mut other_backends = Vec::new();
+
+        for (name, factory) in &self.signing_backend_factories {
+            if config.as_deref() == Some(name) {
+                main_backend = Some(factory(settings));
+                continue;
+            }
+            other_backends.push(factory(settings));
+        }
+
+        Ok(Signer::new(main_backend, other_backends))
+    }
```

Comment:
> Do we plan to store signing backend name somewhere in the repository

I don't think so, no

> to copy the existing store initializer/factory mechanism

I got the idea a long time ago from the custom_backend example, to enable adding new signing backends when using jj as a library.
Right now I'm actually not sure this is really needed, we could keep backend impls internal I guess

Reply:
Ok so I removed the whole factory/initializer stuff, but I still pass around just the signer itself a bunch (which works since as you've mentioned it only depends on the settings) to be able to set a custom signer in tests - and it works for the custom_backend example too